consensus/hotstuff/model/errors.go

```diff
@@ -233,6 +233,18 @@ func IsInvalidSignatureIncludedError(err error) bool {
 	return errors.As(err, &e)
 }
 
+// InvalidAggregatedSignatureError indicates that the aggregated signature is invalid.
```

Comment:
consensus/hotstuff/model and module/signature packages have been using locally defined errors already, that may seem like duplicates. I believe this was a choice made by the consensus team. You can see that for instance with `InsufficientSignaturesError` or `InvalidSignatureIncludedError` that are defined by both packages.

In this PR, I continued using that convention for the newly added `InvalidAggregatedSignatureError`. 
cc @AlexHentschel for visibility.

Reply:
While embedded into the code base, we are trying to implement the consensus such that it is largely decoupled from the remaining code base. This is because we are eventually hoping to isolate the consensus, so academic researchers can run it by itself without needing to spin up a full flow network. Thereby it would become much easier to invite academic contributions, if researchers could directly work on the code base and use it for their consensus experiments. 

Therefore, we are trying to not expose any lower-level errors to the consensus system and instead have a clean abstraction hiding any lower-level platform code such as cryptographic details.